homeassistant/components/icloud/__init__.py

```diff
@@ -1 +1,607 @@
+            device.lost_device(number, message)
+
+    def update_account(service: ServiceDataType) -> None:
+        """Call the update function of an iCloud account."""
+        account = service.data.get(ATTR_ACCOUNT)
+
+        if account is None:
+            for account in hass.data[DOMAIN].values():
+                account.keep_alive()
+        else:
+            _get_account(account).keep_alive()
+
+    def reset_account(service: ServiceDataType) -> None:
+        """Reset an iCloud account."""
+        account = service.data.get(ATTR_ACCOUNT)
+
+        if account is None:
+            for account in hass.data[DOMAIN].values():
+                account.reset_account()
+        else:
+            _get_account(account).reset_account()
+
+    def _get_account(account_identifier: str) -> any:
+        if account_identifier is None:
+            return None
+
+        icloud_account = hass.data[DOMAIN].get(account_identifier, None)
+        if icloud_account is None:
+            for account in hass.data[DOMAIN].values():
+                if account.name == account_identifier:
+                    icloud_account = account
+
+        if icloud_account is None:
+            raise Exception(
+                "No iCloud account with username or name " + account_identifier
+            )
+        return icloud_account
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_ICLOUD_PLAY_SOUND, play_sound, schema=SERVICE_SCHEMA_PLAY_SOUND
+    )
+
+    hass.services.async_register(
+        DOMAIN,
+        SERVICE_ICLOUD_DISPLAY_MESSAGE,
+        display_message,
+        schema=SERVICE_SCHEMA_DISPLAY_MESSAGE,
+    )
+
+    hass.services.async_register(
+        DOMAIN,
+        SERVICE_ICLOUD_LOST_DEVICE,
+        lost_device,
+        schema=SERVICE_SCHEMA_LOST_DEVICE,
+    )
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_ICLOUD_UPDATE, update_account, schema=SERVICE_SCHEMA
+    )
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_ICLOUD_RESET, reset_account, schema=SERVICE_SCHEMA
+    )
+
+    return True
+
+
+class IcloudAccount:
+    """Representation of an iCloud account."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        username: str,
+        password: str,
+        account_name: str,
+        max_interval: int,
+        gps_accuracy_threshold: int,
+    ):
+        """Initialize an iCloud account."""
+        self.hass = hass
+        self._username = username
+        self._password = password
+        self._name = account_name or slugify(username.partition("@")[0])
+        self._fetch_interval = max_interval
+        self._max_interval = max_interval
+        self._gps_accuracy_threshold = gps_accuracy_threshold
+
+        self.api = None
+        self._owner_fullname = None
+        self._family_members_fullname = {}
+        self._devices = {}
+
+        self.unsub_device_tracker = None
+
+    def reset_account(self):
```

Comment:
I don't see any resetting happening ?

Reply:
Actually, I was keeping the old `reset_account_icloud` function with its service, but I don't see a use case either.

So I've looked into it since your review, it was created 3 years ago in this pr #4081 described as "it will restart the API. This is useful when not all devices are found at startup or when you have added a new device to your account.".

So yep, no longer needed with my `update_devices` function now.